_posts/en/newsletters/2022-05-04-newsletter.md

```diff
@@ -0,0 +1,188 @@
+  discussed, but all appeared to have problems, such as [technical
+  challenges][aronesty signal parse scripts], significantly
+  [reducing][grant signal chainalysis] user privacy, [favoring][tetrud
+  signal favor] certain parts of the Bitcoin economy over others, or
+  [penalizing][ivgi signal hodl voting] early voters over those who
+  waited to participate in consensus formation.
+
+    As on previous occasions where this topic has been discussed, it
+    did not appear that any of the suggested methods would produce a
+    result that would be sufficiently respected by most of the
+    discussion participants to strongly inform their decisions about
+    changing Bitcoin's consensus rules.
+
+- **LN anchor outputs security issue:** Bastien Teinturier
+  [posted][teinturier security] to the Lightning-Dev mailing list the
+  announcement of a security issue he had previously responsibly
+  disclosed to LN implementation maintainers.  The issue affected older
+  versions of Core Lightning (with experimental features enabled) and
+  LND.
+
+    Prior to the implementation of [anchor outputs][topic anchor
+    outputs], revoked [HTLC][topic HTLC] transactions only contained a
+    single output, so many implementations only tried to claim that
+    single output.  The design of anchor outputs for LN allows combining
+    multiple revoked HTLC outputs into a single transaction, but this
+    is only safe if implementations claim all of the relevant outputs in
+    the transaction.  Teinturier's implementation of anchor outputs for
+    Eclair allowed him to test the other LN implementations and discover
+    the vulnerability.
+
+    As with a previous attack related to anchor outputs (see [Newsletter
+    #115][news115 fee stealing]), the problem appears to be related to
+    adding support for signing with
+    `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` while still retaining legacy
+    support for signing with `SIGHASH_ALL`.
+
+- **LN gossip rate limiting:** Alex Myers [posted][myers recon] to the
+  Lightning-Dev mailing list about his research into using
+  [minisketch][topic minisketch]-based set reconciliation to reduce the
+  amount of bandwidth nodes need to learn about updates to the LN
+  channel graph.  His method assumes all peers have almost all of the
+  same information about almost all of the same public channels.  One
+  peer can then generate a minisketch from its complete graph of the
+  public network and send that to all of its peers, who can use the
+  minisketch to find any updates to the network since its last
+  reconciliation.  This is different than the proposed use of minisketch
+  for the Bitcoin P2P network via the [erlay][topic erlay] protocol
+  where only updates (new unconfirmed transactions) from the last few
+  seconds are sent.
+
+    One challenge of reconciling over all public channels is that it
+    requires all LN nodes keep the same information.  Any filtering that
```

Comment:
```suggestion
    requires all LN nodes to keep the same information.  Any filtering that
```

Reply:
I think this is better without the preposition, but let me know if there's some rule I'm missing and I'll do better in the future.